src/web/app/user-join-page.component.ts

```diff
@@ -97,9 +98,18 @@ export class UserJoinPageComponent implements OnInit {
     this.courseService.joinCourse(this.key, this.entityType).subscribe(() => {
       this.navigationService.navigateByURL(this.router, `/web/${this.entityType}`);
     }, (resp: ErrorMessageOutput) => {
-      const modalRef: any = this.ngbModal.open(ErrorReportComponent);
-      modalRef.componentInstance.requestId = resp.error.requestId;
-      modalRef.componentInstance.errorMessage = resp.error.message;
+      const errorMessage = resp.error.message;
+
+      if (errorMessage.match(/Course .* is deleted/)) {
+        const modalRef: any = this.ngbModal.open(SimpleModalComponent);
+        modalRef.componentInstance.content = 'The course you are trying to join has been deleted by an instructor';
+        modalRef.componentInstance.isInformationOnly = true;
+        modalRef.componentInstance.confirmMessage = 'OK';
```

Comment:
Should be using `SimpleModalService` here. This is a text-only modal with simple interaction, unlike the error report component.